osu.Game.Rulesets.Mania/Difficulty/Skills/Strain.cs

```diff
@@ -4,98 +4,281 @@
+ * │ Column Strain 0   │  .  ├───f───►  .  ├──────────┼─────────────┼─────────────┼──────f───►  .  ├───►
+ * ├───────────────────┴─────┤       └──┬──┘          │             │             │          └──┬──┘
+ * ├───────────────────┬─────┤          │          ┌──▼──┐          │          ┌──▼──┐          │
+ * │ Column Strain 1   │  .  ├──────────┼──────f───►  .  ├──────────┼──────f───►  .  ├──────────┼──────►
+ * ├───────────────────┴─────┤          │          └──┬──┘          │          └──┬──┘          │
+ * ├───────────────────┬─────┤          │             │          ┌──▼──┐          │             │
+ * │ Column Strain 2   │  .  ├──────────┼─────────────┼──────f───►  .  ├──────────┼─────────────┼──────►
+ * └───────────────────┴─────┘          │             │          └──┬──┘          │             │
+ *                                      │             │             │             │             │
+ * ┌─────────────────────────┐       ┌──▼──┐       ┌──▼──┐       ┌──▼──┐       ┌──▼──┐       ┌──▼──┐
+ * │ Final Strain            │       │  .  │       │  .  │       │  .  │       │  .  │       │  .  │
+ * └─────────────────────────┘       └─────┘       └─────┘       └─────┘       └─────┘       └─────┘
+ *
+ * A few things to notice.
+ * 1) Final Strains are independent of each other.
+ * 2) Global Strain updates on every note, while Column Strain updates only for their respective columns
+ * 3) Final Strain is the summation of global and column strain.
+ */
+
 namespace osu.Game.Rulesets.Mania.Difficulty.Skills
 {
     public class Strain : StrainDecaySkill
     {
-        private const double individual_decay_base = 0.125;
-        private const double overall_decay_base = 0.30;
+        /// <summary> Individual Strain Decay Exponent Base. Used in <see cref="applyDecay"/> </summary>
+        private const double decay_base = 0.125;
+
+        /// <summary> Global Strain Decay Exponent Base. Used in <see cref="applyDecay"/> </summary>
+        private const double global_decay_base = 0.30;
+
+        /// <summary> Center of our endOnBodyBias sigmoid function. </summary>
         private const double release_threshold = 24;
 
         protected override double SkillMultiplier => 1;
         protected override double StrainDecayBase => 1;
 
-        private readonly double[] startTimes;
-        private readonly double[] endTimes;
-        private readonly double[] individualStrains;
+        /// <summary> Previous notes' start times. Indices correspond to columns </summary>
+        private readonly double[] prevStartTimes;
+
+        /// <summary> Previous notes' end times. Indices correspond to columns </summary>
+        private readonly double[] prevEndTimes;
+
+        /// <summary> Per-Column Strains. Indices correspond to columns </summary>
+        private readonly double[] columnStrains;
+
+        /// <summary> Previous Strain processed. </summary>
+        private double prevColumnStrain;
```

Comment:
The xmldoc is actually _worse_ than the variable name here. Only from the variable name can I deduce that this is a _column_ strain and as such _separate_ from global strains.